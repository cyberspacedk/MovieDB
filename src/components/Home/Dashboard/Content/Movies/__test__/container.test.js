import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import MoviesContainer from '../container';

describe('Container: MoviesContainer', () => {
  const store = configureStore()({
    search: {
      ids: [1, 2, 3],
      page: 3,
      total_results: 10,
      loading: false,
      error: false,
      query: 'some film',
    },
    database: {
      movies: {
        1: { id: 1 },
      },
    },
  });

  const wrapper = shallow(<MoviesContainer store={store} />);
  const container = wrapper.dive();

  it('Snapshot: should match', () => {
    expect(container).toMatchSnapshot();
  });
});
