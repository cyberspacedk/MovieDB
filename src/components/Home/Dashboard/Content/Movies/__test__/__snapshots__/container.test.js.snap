// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: MoviesContainer Snapshot: should match 1`] = `
<withRouter(MoviesContainer)
  currentPage={3}
  empty={false}
  error={false}
  loading={false}
  movies={
    Array [
      Object {
        "id": 1,
      },
      undefined,
      undefined,
    ]
  }
  query="some film"
  searchRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  totalResults={10}
/>
`;
